From 30f54666ae15128f26fbad787a35253885a10513 Mon Sep 17 00:00:00 2001
From: Ramya Gnanasekar <rgnanase@codeaurora.org>
Date: Fri, 25 Dec 2020 16:11:06 +0530
Subject: [PATCH] ath11k: Disable rx_header tlv for 2K SKB

On low memory platform hdr_status in hal_rx_desc is not subscribed to
get a savings of 128bytes in skb. This is required to reduce the skb
size from 4K to 2K. Use HTT_H2T_MSG_TYPE_RX_RING_SELECTION_CFG message
to unsubscribe rx_pkt_header tlv for rxdma ring.

Signed-off-by: Ramya Gnanasekar <rgnanase@codeaurora.org>

--- a/drivers/net/wireless/ath/ath11k/dp.h
+++ b/drivers/net/wireless/ath/ath11k/dp.h
@@ -220,6 +220,11 @@ struct ath11k_pdev_dp {
 #define DP_REO_STATUS_RING_SIZE		2048
 #define DP_RXDMA_BUF_RING_SIZE		4096
 #define DP_RXDMA_REFILL_RING_SIZE	2048
+#ifdef CPTCFG_ATH11K_MEM_PROFILE_512MB
+#define DP_RXDMA_NSS_REFILL_RING_SIZE	1816
+#else
+#define DP_RXDMA_NSS_REFILL_RING_SIZE	2048
+#endif
 #define DP_RXDMA_ERR_DST_RING_SIZE	1024
 #define DP_RXDMA_MON_STATUS_RING_SIZE	ATH11K_DP_RXDMA_MON_STATUS_RING_SIZE
 #define DP_RXDMA_MONITOR_BUF_RING_SIZE	ATH11K_DP_RXDMA_MONITOR_BUF_RING_SIZE
--- a/drivers/net/wireless/ath/ath11k/nss.c
+++ b/drivers/net/wireless/ath/ath11k/nss.c
@@ -1802,7 +1802,7 @@ static int ath11k_nss_init(struct ath11k

 	/* fill rx parameters to initialize rx context */
 	wim->wrip.tlv_size = ab->hw_params.hal_desc_sz;
-	wim->wrip.rx_buf_len = DP_RX_BUFFER_SIZE;
+	wim->wrip.rx_buf_len = DP_RXDMA_NSS_REFILL_RING_SIZE;
 	if (of_property_read_bool(dev->of_node, "nss-radio-priority"))
 		wim->flags |= WIFILI_MULTISOC_THREAD_MAP_ENABLE;

